src/main/java/io/phasetwo/service/resource/MembersResource.java

```diff
@@ -36,12 +36,15 @@ public MembersResource(OrganizationAdminResource parent, OrganizationModel organ
   public Stream<UserRepresentation> getMembers(
       @QueryParam("search") String searchQuery,
       @QueryParam("first") Integer firstResult,
-      @QueryParam("max") Integer maxResults) {
+      @QueryParam("max") Integer maxResults,
+      @QueryParam("includeServiceAccounts") Boolean includeServiceAccounts) {
```

Comment:
Could we please add a test with for this flag? You can use the test testAddGetDeleteMemberships as a inspiration

Reply:
@kedare Please add a test for this flag.

Reply:
I am checking the test, I guess it's the `testSearchMembersWithMultipleNameParameter`, I don't think there is any helper to add a service account ? Looks like `createUser` would only be used to create standard users, how should I proceed ?

Reply:
@xgp ping if you have time to check the comments, I can't go further so far without more information, I am not super familiar with Java testing

Reply:
@rtufisi can you help @kedare with a test for this method?

Reply:
Sure. I will help

Reply:
Hei @kedare . Please add the following to your commit

in the OrganizationResourceTest.class

` @Test
  void testIncludeServiceAccountsQueryParameter() throws IOException {
    OrganizationRepresentation org = createDefaultOrg();
    String id = org.getId();

    Response response = getRequest(id, "members");
    assertThat(response.statusCode(), is(Status.OK.getStatusCode()));

    // create a user
    UserRepresentation user1 = createUser(keycloak, REALM, "johndoe");
    UserRepresentation user2 = createUser(keycloak, REALM, "johndow");
    UserRepresentation user3 = createServiceAccountClient(keycloak, REALM, "client1");
    UserRepresentation user4 = createServiceAccountClient(keycloak, REALM, "client2");

    // add membership
    response = putRequest("foo", org.getId(), "members", user1.getId());
    assertThat(response.getStatusCode(), is(Status.CREATED.getStatusCode()));
    response = putRequest("foo", org.getId(), "members", user2.getId());
    assertThat(response.getStatusCode(), is(Status.CREATED.getStatusCode()));
    response = putRequest("foo", org.getId(), "members", user3.getId());
    assertThat(response.getStatusCode(), is(Status.CREATED.getStatusCode()));
    response = putRequest("foo", org.getId(), "members", user4.getId());
    assertThat(response.getStatusCode(), is(Status.CREATED.getStatusCode()));

    // search members with query parameter
    response = getRequest(id, "members?includeServiceAccounts=true");
    assertThat(response.statusCode(), is(Status.OK.getStatusCode()));
    List<UserRepresentation> members = objectMapper().readValue(response.getBody().asString(), new TypeReference<>() {});
    assertThat(members, notNullValue());
    assertThat(members, hasSize(5)); // including org admin default
    assertThat(members, hasItem(hasProperty("username", is("johndoe"))));
    assertThat(members, hasItem(hasProperty("username", is("johndow"))));
    assertThat(members, hasItem(hasProperty("username", is(SERVICE_ACCOUNT_PREFIX + "client1"))));
    assertThat(members, hasItem(hasProperty("username", is(SERVICE_ACCOUNT_PREFIX + "client2"))));

    response = getRequest(id, "members?includeServiceAccounts=false");
    assertThat(response.statusCode(), is(Status.OK.getStatusCode()));
    members = objectMapper().readValue(response.getBody().asString(), new TypeReference<>() {});
    assertThat(members, notNullValue());
    assertThat(members, hasSize(3)); // including org admin default
    assertThat(members, hasItem(hasProperty("username", is("johndoe"))));
    assertThat(members, hasItem(hasProperty("username", is("johndow"))));

    // delete user
    deleteUser(keycloak, REALM, user1.getId());
    deleteUser(keycloak, REALM, user2.getId());
    deleteClient("client1");
    deleteClient("client2");

    // delete org
    deleteOrganization(id);
  }` 

in the Helpers.class:
- at the begining of the class add the constant:
 public static final String SERVICE_ACCOUNT_PREFIX = "service-account-";
- add the helper class:
 `  public static UserRepresentation createServiceAccountClient(
          Keycloak keycloak, String realm, String clientId) {
    ClientRepresentation clientRepresentation = new ClientRepresentation();
    clientRepresentation.setEnabled(true);
    clientRepresentation.setClientId(clientId);
    clientRepresentation.setServiceAccountsEnabled(true);
    keycloak.realm(realm).clients().create(clientRepresentation);

    return keycloak.realm(realm).users().searchByUsername(SERVICE_ACCOUNT_PREFIX+ clientId, true).get(0);
  }`


Please let me know if all works